src/praisonai-agents/praisonaiagents/secrets.py

```
@@ -0,0 +1,289 @@
+    def _resolve_exec(ref: SecretRef) -> SecretResolution:
+        import shlex
+        import subprocess
+
+        try:
+            argv = shlex.split(ref.id)
+        except ValueError as exc:
+            return SecretResolution(UNAVAILABLE, detail=f"bad command {ref.id!r}: {exc}")
+        if not argv:
+            return SecretResolution(MISSING, detail="empty command")
+        try:
+            proc = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)